com.unity.render-pipelines.high-definition/Runtime/Sky/HDRISky/HDRISky.cs

```diff
@@ -169,6 +169,21 @@ public override int GetHashCode()
             return hash;
         }
 
+        /// <summary>
+        /// Determines if the SkySettings is very different from another. This is going to be used to determine whether
+        /// to reset completely the ambient probe instead of using previous one when waiting for current data upon changes.
+        /// In addition to the checks done with the base function, this HDRISky override checks whether the cubemap parameter
+        /// has changed if both settings are HDRISky.
+        /// </summary>
+        /// <param name="otherSettings">The settings to compare with.</param>
+        /// <returns>Whether the settings are deemed very different.</returns>
+        public override bool IsVeryDifferentFrom(SkySettings otherSettings)
```

Comment:
I am just worried that we introduce a public API name "IsVeryDifferentFrom". Summoning @JordanL8  in case he have an idea of a better name :) 

Should we have instead the notion of threshold. Like :   NeedRecomputeAmbientProbe(SkySettings otherSettings) or a name like that?  haveSettingsChange

Reply:
Yeah I can't say that I'm a fan of IsVeryDifferentFrom. The summary says its just used to determine whether to reset the ambient probe, but can it be used for other things? If so, I'd say stick to the is different route but call it something like **DivergesFrom**.

Reply:
The problem with NeedRecomputeAmbientProbe is that it makes no sense in the context of comparing skySettings without having the otherSkySetting being exactly the previous sky setting for the same sky. However that cannot be enforced at this level. 

What we are just saying with that function is that those are different, the summary mentions ambient probe because that's how we use it, but it is mentioned as a note for our use. 

I can put DivergesFrom, but that is not correct either as even with a small difference we have a divergence. This is true only if there is a big divergence. 

I am open to better names :D

Reply:
Are you just open or are you very open? :p Although **Very** tells the users its more than a small difference, they'd still have to look at the docs to find out what Very actually means (like what threshold for what properties etc...). Maybe just ignore Very and mention in the summary what the sky settings need to be in order to be different. That way the information is in the most visible place because the summary shows up in the user's IDE when they type the method.

Reply:
But not everyone reads documentation :D If I have .DivergesFrom(other) returning false for something that diverges less than a certain amount I'd be very confused and angry D:

Reply:
True, I guess with very there they would have to look up the docs to figure out what it means, but without it, they could assume its just like one property and get angry when it isn't as you say. How about Significantly? SignificantlyDivergesFrom or IsSignificantlyDifferentFrom?

Reply:
This I like! Will change